.github/workflows/test.yml

```diff
@@ -35,49 +33,34 @@ jobs:
 
     name: "Test: PHP ${{ matrix.php }}"
     continue-on-error: ${{ matrix.php == '8.2' }}
+    runs-on: ubuntu-latest
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Set coverage variable
-        id: set_cov
-        run: |
-          if [ ${{ matrix.coverage }} == "true" ]; then
-            echo '::set-output name=COV::xdebug'
-          else
-            echo '::set-output name=COV::none'
-          fi
-
-      - name: Install PHP
+      - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
           ini-values: error_reporting=-1, display_errors=On, log_errors_max_len=0
-          coverage: ${{ steps.set_cov.outputs.COV }}
+          coverage: ${{ matrix.coverage && 'xdebug' || 'none' }}
           tools: cs2pr
 
       # At least one test needs a non-en_US locale to be available, so make sure it is.
       - name: Install locales
         run: |
           sudo apt-get update
-          sudo apt-get install locales-all
+          sudo apt-get -y install locales-all
 
       - name: Show available locales
         run: locale -a
 
       # Install dependencies and handle caching in one go.
-      # @link https://github.com/marketplace/actions/install-composer-dependencies
-      - name: Install Composer dependencies - normal
-        if: ${{ matrix.php != '8.2' }}
-        uses: "ramsey/composer-install@v2"
-
-      # For PHP "nightly", we need to install with ignore platform reqs.
       - name: Install Composer dependencies - with ignore platform
-        if: ${{ matrix.php == '8.2' }}
         uses: "ramsey/composer-install@v2"
         with:
-          composer-options: --ignore-platform-req=php
+          composer-options: ${{ matrix.php == '8.2' && '--ignore-platform-req=php' || '' }}
```

Comment:
One step will do.

Reply:
See above. While it works, it obfuscates what's happening and also makes the output of the script less readable.

With the separate steps, you can see straight away when opening a build result which of the two versions of `composer install` where run, without having to fold out the step details. This is no longer the case in your version, making the debugging of builds more cumbersome.